pkg/filter/driftignore.go

```diff
@@ -39,36 +39,40 @@ func (r *DriftIgnore) readIgnoreFile() error {
 		line := scanner.Text()
 		typeVal := readDriftIgnoreLine(line)
 		nbArgs := len(typeVal)
-		if nbArgs < 2 {
-			logrus.WithFields(logrus.Fields{
-				"line": line,
-			}).Warnf("unable to parse line, invalid length, got %d expected >= 2", nbArgs)
-			continue
-		}
-		if nbArgs == 2 { // We want to ignore a resource (type.id)
+		if nbArgs == 0 || strings.HasPrefix(line, "#") {
```

Comment:
`nbArgs == 0` is not exactly the check we want to ignore, we want to check if the line is empty

Your if directive should be moved just after `line := scanner.Text()`.
We do not want to call `readDriftIgnoreLine` for ignored lines.

You can also remove the `else` directive as you put a continue.

Something like below should do the job :

```diff
line := scanner.Text()
+if line == "" || strings.HasPrefix(line, "#") {
+  logrus.WithFields(logrus.Fields{
+    "line": line,
+  }).Debug("Skipped comment or empty line")
+  continue
+}
typeVal := readDriftIgnoreLine(line)
```